File: Python/ui.py
from forest import Forest, ForestRepository
from basket import Basket


class UI:
    def get_mushroom_from_user(self, forest: Forest, repo: ForestRepository):
        name = input("Enter mushroom name: ")
        eatable = input("Is mushroom eatable? (1/0): ")
        treatment = input("Enter heat treatment time for mushroom in minutes: ")
        repo.add_mushroom_to_forest(name, eatable, treatment, forest)

    def get_basket_from_user(self, basket: Basket, forest: Forest):
        mushroom_list = forest.get_mushrooms()
        print("\nEnter 'd' if done\nEnter mushrooms:\n")
        while True:
            name = input()
            if name == 'd':
                break
            else:
                counter = 0
                for mushroom in mushroom_list:
                    if name == mushroom.get_name():
                        basket.add_item(mushroom)
                        counter += 1
                        break
                if counter == 0:
                    print("Mushroom not found")
        basket.calculate_result()

    def show_mushrooms(self, forest: Forest):
        print("NAME  |  EATABLE  |  TREATMENT")
        for mushroom in forest.get_mushrooms():
            print(mushroom)

    def show_menu(self):
        print("\n1. Add mushroom")
        print("2. Show mushrooms in forest")
        print("3. Calculate eatable percent and treatment time")
        print("4. Exit")

File: C++/OOP Lab3/ui.cpp
#include "ui.h"

void UI::getMushroomFromUser(Forest& forest, ForestRepository& repo) {
	system("cls");
	std::string name;
	bool eatable;
	int treatment;

	std::cout << "Enter mushroom name: ";
	std::cin >> name;
	std::cout << "Is mushroom eatable? (1/0): ";
	std::cin >> eatable;
	std::cout << "Enter heat treatment time for mushroom in minutes: ";
	std::cin >> treatment;

	repo.addMushroomToForest(name, eatable, treatment, forest);
}

void UI::getBasketFromUser(Basket& basket, Forest& forest) {
	system("cls");
	std::vector<Mushroom*> mushroomList = forest.getMushrooms();
	std::cout << "Enter 'd' if done" << std::endl << "Enter mushroom:" << std::endl;

	while (true) {
		std::string name;
		std::cin >> name;
		if (name == "d") {
			break;
		}
		else {
			int counter = 0;
			for (auto mushroom : mushroomList) {
				if (name == mushroom->getName()) {
					basket.addItem(mushroom);
					counter++;
					break;
				}
			}
			if (counter == 0) {
				std::cout << "Mushroom not found" << std::endl;
			}
		}
	}
	basket.calculateResult();
}

void UI::showMushrooms(Forest& forest) {
	system("cls");
	std::cout << "NAME  |  EATABLE  |  TREATMENT" << std::endl;
	for (auto mushroom : forest.getMushrooms()) {
		std::cout << *mushroom << std::endl;
	}
	system("pause");
}

void UI::showMenu() {
	system("cls");
	std::cout << "1. Add mushroom" << std::endl;
	std::cout << "2. Show mushrooms in forest" << std::endl;
	std::cout << "3. Calculate eatable percent and treatment time" << std::endl;
	std::cout << "4. Exit" << std::endl;
}